### PR TITLE
feat(CacaX): add presence

### PR DESCRIPTION
**Pull request by DorafBG**
## Description 
- New presence for the social media [CacaX](https://cacax.fun).
- Shows the name of the user who you are watching. You can go to this user profile with a button "Voir le profil".
- Shows the name of the user of the post which you are watching. You can go to this post with a button "Voir le profil".
- Shows the user avatar when we are on a post or on a profile.
- When we are on the home page, the messages page, about us page, privacy page, this is shown on the presence.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image1

image2

image3

image4

image5

image6

image7

</details>

### CODE REVIEW

File: websites/C/CacaX/metadata.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"id": "418760799226429441",
+		"name": "doraf"
+	},
+	"service": "CacaX",
+	"description": {
+		"en": "The social network of the future, 100% free of expression !"
+	},
+	"url": "cacax.fun",
```
Comment (theusaf) [inline websites/C/CacaX/metadata.json:11]:
```suggestion
	"url": [
		"www.cacax.fun",
		"cacax.fun"
	],
```

File: websites/C/CacaX/presence.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			})
+			.join(" ");
+	};
+
+	presence.on("UpdateData", async () => {
+		const { pathname, href } = document.location;
+
+		const presenceData: PresenceData = {
+			details: "",
+			state: "",
```
Comment (theusaf) [inline websites/C/CacaX/presence.ts:19]: Empty fields are not necessary.
```suggestion
```

File: websites/C/CacaX/presence.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		} else if (pathname === "/messages") {
+			presenceData.details = "Messagerie";
+			presenceData.state = "Regarde ses messages...";
+
+		} else if (pathname.includes("/users/")) {
+			const userArobase = document.querySelector("span").textContent;
+			presenceData.details = "Profil de : " + userArobase;
+			const userName = document.querySelector("h1").textContent;
+			presenceData.state = `Regarde le profil de l'utilisateur ${userName}...`;
+			presenceData.buttons = [{label: "Voir le profil", url: href}];
```
Comment (theusaf) [inline websites/C/CacaX/presence.ts:47]: Button labels should be capitalized
```suggestion
			presenceData.buttons = [{label: "Voir le Profil", url: href}];
```

File: websites/C/CacaX/presence.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+				if (!isInsideExcludedDiv && images[i].src.startsWith("data:image/") && images[i].alt === "Avatar") {
+					imgUrl = images[i].src;
+					presenceData.smallImageKey = imgUrl;
+					break;  
+				}
+			}
+			
+			// si rien trouvé
+			if (imgUrl == "") {
+				presenceData.largeImageKey = "https://www.cacax.fun/_next/image?url=%2Fimages%2Fplaceholder.png&w=1920&q=75";
```
Comment (theusaf) [inline websites/C/CacaX/presence.ts:66]:
```suggestion
				presenceData.largeImageKey = "https://www.cacax.fun/images/placeholder.png";
```

File: websites/C/CacaX/presence.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+		} else if (pathname === "/privacy") {
+			presenceData.details = "Politique de confidentialité";
+			presenceData.state = "Regarde la politique de confidentialité (il veur voir si on respecte le RGPD)...";
+
+		} else if (pathname.includes("/posts/")) {
+			presenceData.details = "Posts";
+			const userName = document.querySelector("span").textContent;
+			presenceData.state = `Regarde le post de ${userName}...`;
+			presenceData.buttons = [{label: "Voir le post", url: href}];
```
Comment (theusaf) [inline websites/C/CacaX/presence.ts:84]:
```suggestion
			presenceData.buttons = [{label: "Voir le Post", url: href}];
```

**Review by theusaf — CHANGES_REQUESTED** [review]
Also make sure to resolve the failed CI checks

**Review by DorafBG — COMMENTED** [review]
thank you for your comments !

**Comment by DorafBG** [comment]
I dont understand how do we fix the presence validator ..

File: websites/C/CacaX/metadata.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
```
Comment (theusaf) [inline websites/C/CacaX/metadata.json:2]:
```suggestion
	"$schema": "https://schemas.premid.app/metadata/1.9",
```

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**